synapse/tests/test_cortex.py

```diff
@@ -296,232 +274,239 @@ def onAdd(node):
+            nodes = await alist(await core.eval('', opts=opts))
 
             self.len(1, nodes)
             self.eq('visi', nodes[0][0][1])
 
-            core.addFeedData('com.test.record', data)
+            await core.addFeedData('com.test.record', data)
 
             # test the remote storm result counting API
-            self.eq(0, core.count('pivtarg'))
-            self.eq(1, core.count('inet:user'))
+            self.eq(0, await core.count('pivtarg'))
+            self.eq(1, await core.count('inet:user'))
 
-    def test_stormcmd(self):
+    async def test_stormcmd(self):
 
-        with self.getTestCore() as core:
+        async with self.getTestDmon(mirror='dmoncore') as dmon, \
+                await self.agetTestProxy(dmon, 'core') as core:
 
-            msgs = list(core.storm('|help'))
+            msgs = await alist(await core.storm('|help'))
             self.printed(msgs, 'help: List available commands and a brief description for each.')
 
-            msgs = list(core.storm('help'))
+            msgs = await alist(await core.storm('help'))
             self.printed(msgs, 'help: List available commands and a brief description for each.')
 
-            list(core.eval('[ inet:user=visi inet:user=whippit ]'))
+            await alist(await core.eval('[ inet:user=visi inet:user=whippit ]'))
 
-            self.len(2, list(core.eval('inet:user')))
+            await self.agenlen(2, await core.eval('inet:user'))
 
             # test cmd as last text syntax
-            self.len(1, list(core.eval('inet:user | limit 1')))
+            await self.agenlen(1, await core.eval('inet:user | limit 1'))
 
             # test cmd and trailing pipe syntax
-            self.len(1, list(core.eval('inet:user | limit 1|')))
+            await self.agenlen(1, await core.eval('inet:user | limit 1|'))
 
             # test cmd and trailing pipe and whitespace syntax
-            self.len(1, list(core.eval('inet:user | limit 1    |     ')))
+            await self.agenlen(1, await core.eval('inet:user | limit 1    |     '))
 
             # test cmd and trailing pipe and whitespace syntax
-            self.len(2, list(core.eval('inet:user | limit 10 | [ +#foo.bar ]')))
-            self.len(1, list(core.eval('inet:user | limit 10 | +inet:user=visi')))
+            await self.agenlen(2, await core.eval('inet:user | limit 10 | [ +#foo.bar ]'))
+            await self.agenlen(1, await core.eval('inet:user | limit 10 | +inet:user=visi'))
 
-            # test invalid option sytnax
-            msgs = list(core.storm('inet:user | limit --woot'))
+            # test invalid option syntax
+            msgs = await alist(await core.storm('inet:user | limit --woot'))
             self.printed(msgs, 'usage: limit [-h] count')
             self.len(0, [m for m in msgs if m[0] == 'node'])
 
-    def test_onsetdel(self):
+    async def test_onsetdel(self):
 
-        with self.getTestCore() as core:
+        args_hit = None
```

Comment:
can make this a dictionary and make the callback just set keys in the dict and avoid nonlocal use

Reply:
It is a general patterned we've used in other tests and avoided locality related keywords as a result

Reply:
ok